libs/messenger-bundle/src/ConnectionEvent/ApplicationEvent/Storage/ComponentConfigurationDeletedEvent.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\MessengerBundle\ConnectionEvent\ApplicationEvent\Storage;
+
+use InvalidArgumentException;
+use Keboola\MessengerBundle\ConnectionEvent\EventInterface;
+
+class ComponentConfigurationDeletedEvent implements EventInterface
+{
+    public const NAME = 'storage.componentConfigurationDeleted';
+
+    public function __construct(
+        public readonly string $uuid,
+        public readonly int $projectId,
+
+        public readonly int $accessTokenId,
+        public readonly string $accessTokenName,
+
+        public readonly string|int $objectId,
+        public readonly string $objectType,
+        public readonly string $objectName,
+
+        public readonly string $message,
+        public readonly array $params,
+
+        public readonly int $idBranch,
+    ) {
+    }
+
+    public static function fromArray(array $data): self
+    {
+        if ($data['name'] !== self::NAME) {
+            throw new InvalidArgumentException(sprintf(
+                '%s expects event name "%s" but is "%s"',
+                static::class,
+                self::NAME,
+                $data['name'],
+            ));
+        }
+
+        return new self(
+            $data['uuid'],
+            $data['idProject'],
+            $data['idAccessToken'],
+            $data['accessTokenName'],
+            $data['objectId'],
+            $data['objectType'],
+            $data['objectName'],
+            $data['message'],
+            $data['params'],
+            $data['idBranch'],
+        );
+    }
+
+    public function toArray(): array
+    {
+        return [
+            'name' => self::NAME,
+            'uuid' => $this->uuid,
+            'idProject' => $this->projectId,
+            'idAccessToken' => $this->accessTokenId,
+            'accessTokenName' => $this->accessTokenName,
+            'objectId' => $this->objectId,
+            'objectType' => $this->objectType,
+            'objectName' => $this->objectName,
+            'message' => $this->message,
+            'params' => $this->params,
+            'idBranch' => $this->idBranch,
+        ];
+    }
+
+    public function getId(): string
+    {
+        return $this->uuid;
+    }
+
+    public function getEventName(): string
+    {
+        return self::NAME;
+    }
+
+    public function getComponent(): string
+    {
+        return $this->params['component'];
+    }
+
+    public function getConfigurationId(): string
+    {
+        return $this->params['configurationId'];
+    }
```

Comment:
Tady asi nebude nenovejsi PHPStan 🙂 Ty `params` nemas nijak validovany. Bud to vytahni do separatnich properties a namapuj ve `fromArray` (ktery je pri deserializaci obaleny `try/catch(Throwable)`, takze to aspon nejak chyti. Nebo nech v params, ale ve `fromArray` je nejak zvaliduj (`if(...) throw InvalidArgumentException`) a `params` property popis phpdocem.

Reply:
ok

Reply:
Toho obaleni v try catchi pri `fromArray` jsem si nevsiml, a dal jsem to nezkoumal. Predelal jsem to teda na ty properties (navic v audit logu eventac je to podobne jsem koukal)